app/src/main/java/cm/aptoide/pt/home/AdMapper.java

```diff
@@ -1,42 +1,56 @@
 package cm.aptoide.pt.home;
 
-import cm.aptoide.pt.dataprovider.model.v2.GetAdsResponse;
 import cm.aptoide.pt.search.model.SearchAdResult;
+import java.util.AbstractMap;
 import rx.functions.Func1;
 
 public class AdMapper {
   public AdMapper() {
   }
 
-  Func1<? super GetAdsResponse.Ad, SearchAdResult> mapAdToSearchAd() {
+  Func1<? super AdClick, AbstractMap.SimpleEntry<String, SearchAdResult>> mapAdToSearchAd() {
     return ad -> {
       if (ad == null) {
-        return new SearchAdResult();
+        return new AbstractMap.SimpleEntry<>("", new SearchAdResult());
       }
       String clickUrl = null;
       int networkId = 0;
-      if (ad.getPartner() != null) {
-        networkId = ad.getPartner()
+      if (ad.getAd()
+          .getPartner() != null) {
+        networkId = ad.getAd()
+            .getPartner()
             .getInfo()
             .getId();
 
-        clickUrl = ad.getPartner()
+        clickUrl = ad.getAd()
+            .getPartner()
             .getData()
             .getClickUrl();
       }
-      return new SearchAdResult(ad.getInfo()
-          .getAdId(), ad.getData()
-          .getIcon(), ad.getData()
-          .getDownloads(), ad.getData()
-          .getStars(), ad.getData()
+      return new AbstractMap.SimpleEntry<>(ad.getTag(), new SearchAdResult(ad.getAd()
+          .getInfo()
+          .getAdId(), ad.getAd()
+          .getData()
+          .getIcon(), ad.getAd()
+          .getData()
+          .getDownloads(), ad.getAd()
+          .getData()
+          .getStars(), ad.getAd()
+          .getData()
           .getModified()
-          .getDate(), ad.getData()
-          .getPackageName(), ad.getInfo()
-          .getCpcUrl(), ad.getInfo()
-          .getCpdUrl(), ad.getInfo()
-          .getCpiUrl(), clickUrl, ad.getData()
-          .getName(), ad.getData()
-          .getId(), networkId);
+          .getDate(), ad.getAd()
+          .getData()
+          .getPackageName(), ad.getAd()
+          .getInfo()
+          .getCpcUrl(), ad.getAd()
+          .getInfo()
+          .getCpdUrl(), ad.getAd()
+          .getInfo()
+          .getCpiUrl(), clickUrl, ad.getAd()
+          .getData()
+          .getName(), ad.getAd()
+          .getData()
+          .getId(), networkId));
```

Comment:
ad.getAd() is used several times, shouldn't it be extracted?

Reply:
extracted